src/main/scala/higherkindness/compendium/http/HttpUtils.scala

```diff
@@ -16,38 +16,20 @@
 
 package higherkindness.compendium.http
 
-import org.http4s.multipart.Multipart
-import fs2.text._
+import cats.implicits._
 import cats.effect.Sync
 import higherkindness.compendium.models.Protocol
 import org.apache.avro.Schema
-import org.http4s.InvalidMessageBodyFailure
 
 final class HttpUtils[F[_]: Sync] {
 
   private val parser: Schema.Parser = new Schema.Parser()
 
-  def filename(multipart: Multipart[F]): F[String] =
-    multipart.parts
-      .find(_.filename.isDefined)
-      .flatMap(_.filename)
-      .fold(Sync[F].raiseError[String](
-        new InvalidMessageBodyFailure("Missing filename from upload.")))(Sync[F].pure)
-
-  import cats.implicits._
-
-  def rawText(multipart: Multipart[F]): F[String] =
-    multipart.parts
-      .map(_.body.through(utf8Decode).compile.foldMonoid)
-      .toList
-      .sequence
-      .map(_.mkString("\n"))
-
-  def protocol(name: String, text: String): F[Protocol] =
-    if (text.trim.isEmpty)
+  def parseProtocol(protocol: Protocol): F[Protocol] =
```

Comment:
Would it be more accurate call this method `validateProtocol` instead of `parseProcotol`?

Reply:
Maybe `validateParseProtocol`? At the end is what this piece of code is doing 🤔